src/main/java/com/bitso/examples/BitsoWebSocketExample.java

```diff
@@ -1,32 +1,212 @@
+        private BigDecimal mMinPrice;
+        private BigDecimal mMaxPrice;
+
+        public OrderStructure(){
+            mPriceMap = new HashMap<BigDecimal, BigDecimal>();
+            mMinPrice = new BigDecimal(0);
+            mMaxPrice = new BigDecimal(0);
+        }
+        
+        public void insertPrice(BigDecimal price, BigDecimal amount){
+            mPriceMap.put(price, amount);
+            
+            if(mMinPrice.compareTo(price) > 0){
+                mMinPrice = price;
+            }
+            
+            if(mMaxPrice.compareTo(price) < 0){
+                mMaxPrice = price;
+            }
+        }
+        
+        public void updatePrice(BigDecimal price, BigDecimal amount){
+            mPriceMap.put(price, amount);
+            updateMaxMin();
+        }
+        
+        public void removePrice(BigDecimal price){
+            mPriceMap.remove(price);
+            
+            if((mMinPrice.compareTo(price) == 0) || (mMaxPrice.compareTo(price) == 0)){
+                updateMaxMin();
+            }
+        }
+        
+        public boolean priceExists(BigDecimal price){
+           return mPriceMap.containsKey(price);
+        }
+        
+        public void updateMaxMin(){
+            ArrayList<BigDecimal> publicOrders = getSortedOrders();
+            mMinPrice = publicOrders.get(0);
+            mMaxPrice = publicOrders.get(publicOrders.size() - 1);
+        }
+        
+        public ArrayList<BigDecimal> getSortedOrders(){
+            ArrayList<BigDecimal> prices = new ArrayList<BigDecimal>();
+            
+            Iterator<Map.Entry<BigDecimal, BigDecimal>> it = mPriceMap.entrySet().iterator();
+            
+            while(it.hasNext()){
+                Map.Entry<BigDecimal, BigDecimal> entry = it.next(); 
+                prices.add(entry.getValue());
+            }
+            Collections.sort(prices);
+            
+            return prices;
+        }
+
+        public BigDecimal getMinPrice() {
+            return mMinPrice;
+        }
+
+        public BigDecimal getMaxPrice() {
+            return mMaxPrice;
         }
-        System.out.println("got a message: " + arg);
     }
 
-    public static void main(String[] args) {
-        new BitsoWebSocketExample();
+    public static void main(String args[]) throws SSLException, URISyntaxException, InterruptedException{
+        final BitsoChannels[] bitsoChannels = { BitsoChannels.TRADES,
+                BitsoChannels.DIFF_ORDERS};
+        
+        BitsoWebSocket bitsoWebSocket = new BitsoWebSocket();
+        BitsoWebSocketExample bitsoWebSocketExample = new BitsoWebSocketExample();
+        
+        bitsoWebSocket.addObserver(bitsoWebSocketExample);
+        
+        bitsoWebSocket.openConnection();
+        for (BitsoChannels bitsoChannel : bitsoChannels) {
+            bitsoWebSocket.subscribeBitsoChannel(bitsoChannel.toString());
+        }
+        
+        bitsoWebSocketExample.getInitialOrderBook();
```

Comment:
Doesn't seem like this follows the algo described in the docs: https://bitso.com/api_info#general41

What happens if you you receive a websocket update before the InitialOrderBook is fetched?